### ISSUE TITLE
resetNavTreeControl crashes MMEX

### ISSUE THREAD

**Issue by slodki** [moneymanagerex/moneymanagerex#1566]
1. Open existing database
2. Click open icon from toolbar and select existing encrypted database
3. Enter **wrong** password when prompted
4. Pop-up is displayed...
5. MMEX crashes with following error:
```
10:48:19: Debug: SQLITE_UPDATE
10:48:19: Debug: database: main, table: INFOTABLE_V1, rowid: 22
10:48:19: Debug: =========== navTreeStateToJson =============================
10:48:19: Debug: {
    "item@Home Page": true
}
10:48:19: Debug: SQLITE_UPDATE
10:48:19: Debug: database: main, table: INFOTABLE_V1, rowid: 22
10:48:20: Debug: =========== navTreeStateToJson =============================
10:48:20: Debug: {
    "item@Home Page": true,
    "item@Bank Accounts": true
}
10:48:20: Debug: SQLITE_UPDATE
10:48:20: Debug: database: main, table: INFOTABLE_V1, rowid: 22
terminate called after throwing an instance of 'wxSQLite3Exception'
```

This error happens while executing `resetNavTreeControl()` in [`mmGUIFrame::SetDatabaseFile()`](https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmframe.cpp#L3046)

BUT this problem not exist in the similar case:
1. Start MMEX
2. Select Open existing database from startup dialog
3. **Cancel** file selection
4. Click open icon from toolbar and select existing encrypted database
5. Enter **wrong** password when prompted
6. Pop-up is displayed...
7. No crash at `resetNavTreeControl()` in [`mmGUIFrame::mmGUIFrame()`](https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmframe.cpp#L293) (because nav tree was empty?)

<hr><small><b>Version</b>: 1.4.0-Alpha.3 • db 7 (aes256cbc) • git fc185e65 (2018-04-18) master<br><b>Libs</b>: wxWidgets 3.1.1 (wxGTK 2.24) • wxSQLite3 4.0.2 (SQLite 3.23.1) • Mongoose 6.11 • RapidJSON 1.1.0 • Lua 5.3.3 • libcurl 7.55.1 • OpenSSL 1.0.2g • zlib 1.2.11 • libidn2 2.0.2 • libpsl 0.18.0 • ( libidn2 2.0.2) • librtmp 2.3<br><b>Build</b>: Apr 19 2018 10:44:56 • CMake 3.9.1 • GNU Make 4.1 • gettext 0.19.8.1 • GCC 7.2.0 • Ubuntu 17.10 'artful'<br><b>OS</b>: Ubuntu 17.10 "artful" • Linux 4.13.0-38-generic x86_64 • KDE Polish (UTF-8) • 1280x1024 24bit 96x96ppi</small>

**Comment by slodki** [comment]
BTW rowid=22 is NAV_TREE_STATUS

**Comment by slodki** [comment]
problem narrowed to [`delete mmTreeItemData`](https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmframe.cpp#L381) in the following sequence:

```
11:56:37: Debug: 
11:56:37: Debug: item@Bank Accounts
11:56:37: Debug: item@Assets
11:56:37: Debug: item@Bills & Deposits
11:56:37: Debug: item@Budgeting
11:56:37: Debug: item@Transaction Report
11:56:37: Debug: report@Daily
terminate called after throwing an instance of 'wxSQLite3Exception'
```

There is something wrong with `mmReportCashFlow::~mmReportCashFlow()` destructor. It crashes while destructing a `std::vector`. This could be related to following comment from `stl_vector.h` file:
```c++
      /**
       *  The dtor only erases the elements, and note that if the
       *  elements themselves are pointers, the pointed-to memory is
       *  not touched in any way.  Managing the pointer is the user's
       *  responsibility.
       */
      ~vector() _GLIBCXX_NOEXCEPT
      { std::_Destroy(this->_M_impl._M_start, this->_M_impl._M_finish,
		      _M_get_Tp_allocator()); }
```

And declaration :
```c++
    typedef std::vector<mmRepeatForecast> forecastVec;
    std::vector<forecastVec> bdForecastVec;
```

**Comment by vomikan** [comment]
In this case mmex crashed here:
https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmframe.cpp#L381
for report@Daily item

**Comment by vomikan** [comment]
Crash fix:
```
        if (iData) {
            mmPrintableBase* rep = iData->get_report();
            if (!rep) {
                delete iData;
            }
        }
```

**Comment by vomikan** [comment]
for every items when ```delete iData``` called the 
```    ~mmTreeItemData()
    {
        if (report_) delete report_;
    }
```
executed. mmex crashed on delete report_

**Comment by vomikan** [comment]
Next crash candidate:
https://github.com/moneymanagerex/moneymanagerex/blob/master/src/reports/reportbase.cpp#L99

**Comment by slodki** [comment]
See above my debugging traces.
In my case `std::~vector()` crashes somewhere in `mmReportCashFlow::~mmReportCashFlow()` called from `~mmTreeItemData()` called from `if (iData) delete iData;` ...

**Comment by vomikan** [comment]
This fix is working like a sharm
![image](https://user-images.githubusercontent.com/6836805/38990055-8970975a-43e1-11e8-9ab3-99b66da6c7a0.png)

**Comment by vomikan** [comment]
But here is bug №2:
In case if one DB opend then other opend - all reports settings for DB №2 will overwrited by settings from DB №1.

**Comment by vomikan** [comment]
I don't know how but we should init ```mmPrintableBase::~mmPrintableBase()``` before other DB opened.

**Comment by slodki** [comment]
fixed by #1569
